src/rsvc/srv.c

```diff
@@ -524,6 +594,54 @@ static struct rdb_cbs rsvc_rdb_cbs = {
 	.dc_stop	= rsvc_stop_cb
 };
 
+static void
+map_distd(void *arg)
+{
+	struct ds_rsvc *svc = arg;
+
+	D_DEBUG(DB_MD, "%s: start\n", svc->s_name);
+	for (;;) {
+		bool	stop;
+		int	rc;
+
+		ABT_mutex_lock(svc->s_mutex);
+		for (;;) {
+			stop = svc->s_map_distd_stop;
+			if (stop)
+				break;
+			if (svc->s_map_dist) {
+				svc->s_map_dist = false;
+				break;
+			}
+			ABT_cond_wait(svc->s_map_dist_cv, svc->s_mutex);
+		}
+		ABT_mutex_unlock(svc->s_mutex);
+		if (stop)
+			break;
+		rc = rsvc_class(svc->s_class)->sc_map_dist(svc);
+		if (rc != 0)
+			/* Try again. */
+			svc->s_map_dist = true;
+		dss_sleep(3000 /* ms */);
```

Comment:
Sounds like a comment is needed. I also changed the logic a bit to sleep only in the retry case. Let's see how that goes.

sc_map_dist is supposed to be synchronous; otherwise we'd give a hint either in the callback name or the callback comment, I think.